pkg/controller/deployment/deployment_controller.go

```diff
@@ -178,3 +180,54 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileDeployment) syncOnJaegerChanges(event handler.MapObject) []reconcile.Request {
+	reconciliations := []reconcile.Request{}
+	nss := map[string]corev1.Namespace{} // namespace cache
+
+	jaeger, ok := event.Object.(*v1.Jaeger)
+	if !ok {
+		return reconciliations
+	}
+
+	deployments := appsv1.DeploymentList{}
+	err := r.client.List(context.Background(), &deployments)
```

Comment:
Question

In the case of the operator watching one or a couple of namespaces (no cluster wide) Should we limit this to those namespaces?

Reply:
I don't think we have a way to restrict the namespaces being observed for injection? the WATCH_NAMESPACE part is which namespaces to check for CRs, not deployments.
@jpkrohling is that also your understanding?

Reply:
Ok, I thought WATCH_NAMESPACE restricts injection too. My bad.

Reply:
I'd expect the client.List to return the deployments for all namespaces where the jaeger-agent service account has access to. Perhaps QE could confirm that.